templates: make node compilation optional (#5954)
paritytech/polkadot-sdk: cumulus/polkadot-omni-node/README.md

# Polkadot Omni Node

This is a white labeled implementation based on [`polkadot-omni-node-lib`](https://crates.io/crates/polkadot-omni-node-lib).
It can be used to start a parachain node from a provided chain spec file. It is only compatible with runtimes that use block
number `u32` and `Aura` consensus.

## Installation

Download & expose it via `PATH`:

```bash
# Download and set it on PATH.
wget https://github.com/paritytech/polkadot-sdk/releases/download/<stable_release_tag>/polkadot-omni-node
chmod +x polkadot-omni-node
export PATH="$PATH:`pwd`"
```

Compile & install via `cargo`:

```bash
# Assuming ~/.cargo/bin is on the PATH
cargo install polkadot-omni-node
```

## Usage

> A basic example for an Omni Node run starts from a runtime which implements the [sp_genesis_builder::GenesisBuilder](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html).

Reviewer: ```suggestion
A basic example for an Omni Node run starts from a runtime which implements the [`sp_genesis_builder::GenesisBuilder`](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html).
```

Author: [37e43c0](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/37e43c03090ee56e14c5df4da2c42945e6bcc08e)

> The interface mandates the runtime to expose a [named-preset](https://docs.rs/staging-chain-spec-builder/6.0.0/staging_chain_spec_builder/#generate-chain-spec-using-runtime-provided-genesis-config-preset).

Reviewer 2: ```suggestion
The interface mandates the runtime to expose a [named-preset](https://docs.rs/staging-chain-spec-builder/latest/staging_chain_spec_builder/#generate-chain-spec-using-runtime-provided-genesis-config-preset).
```

We should be consistent in using `latest`, I suppose?

Author: [62f757c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/62f757c2c0920bb95b93fcd73e700112a33d7eeb)

> 

Author: Added it to the useful links version: [f94402c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/f94402ccf9ad3c2ad22f12b49579fd75ef9551f2)

> ### Install chain-spec-builder

Reviewer 2: ```suggestion
### 1. Install chain-spec-builder
```

Author: [62f757c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/62f757c2c0920bb95b93fcd73e700112a33d7eeb)

**Note**: `chain-spec-builder` binary is published on [crates.io](https://crates.io) under
[staging-chain-spec-builder](https://crates.io/crates/staging-chain-spec-builder) due to a name conflict.
Install it with `cargo` like bellow :

```bash
cargo install staging-chain-spec-builder
```

> ### Generate a chain spec

Reviewer 2: ```suggestion
### 2. Generate a chain spec
```

Author: [62f757c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/62f757c2c0920bb95b93fcd73e700112a33d7eeb)

> It is also expected for the chain spec to contain parachains related fields like `relay_chain` and `para_id`.

Author: [f94402c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/f94402ccf9ad3c2ad22f12b49579fd75ef9551f2)

These fields can be introduced by running [staging-chain-spec-builder](https://crates.io/crates/staging-chain-spec-builder)
with additional flags:

```bash
chain-spec-builder create --relay-chain <relay_chain_id> --para-id <id> -r <runtime.wasm> named-preset <preset_name>
```

> ### Run Omni Node

Reviewer 2: ```suggestion
### 3. Run Omni Node
```

Just a suggestion, I feel like the numbers give a better idea that these are 3 sequential steps

Author: Done here: [62f757c](https://github.com/paritytech/polkadot-sdk/pull/5954/commits/62f757c2c0920bb95b93fcd73e700112a33d7eeb).
